libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/couchdb/AttachmentConnector.java

```diff
@@ -75,26 +77,37 @@ public void deleteAttachment(String id) {
     }
 
     public void deleteAttachments(Collection<Attachment> attachments) {
-        Set<String> attachmentContentIds = getAttachmentContenIds(attachments);
+        Set<String> attachmentContentIds = getAttachmentContentIds(attachments);
         deleteAttachmentsByIds(attachmentContentIds);
     }
 
     private void deleteAttachmentsByIds(Collection<String> attachmentContentIds) {
         connector.deleteIds(attachmentContentIds, AttachmentContent.class);
     }
 
-    private Set<String> getAttachmentContenIds(Collection<Attachment> attachments) {
+    private Set<String> getAttachmentContentIds(Collection<Attachment> attachments) {
         return nullToEmptyCollection(attachments).stream()
                 .map(Attachment::getAttachmentContentId)
                 .collect(Collectors.toSet());
     }
 
-    public void deleteAttachmentDifference(Set<Attachment> before, Set<Attachment> after){
+    public void deleteAttachmentDifference(Set<Attachment> attachmentBefore, Set<Attachment> attachmentAfter){
         // it is important to take the set difference between sets of ids, not of attachments themselves
-        // otherwise, when `after` contains the same attachment (with the same id), but with one field changed (e.g. sha1),
+        // otherwise, when `attachmentAfter` contains the same attachment (with the same id), but with one field changed (e.g. sha1),
         // then they are considered unequal and the set difference will contain this attachment and therefore
-        // deleteAttachments(Collection<Attachment>) will delete an attachment that is present in `after`
-        deleteAttachmentsByIds(Sets.difference(getAttachmentContenIds(before), getAttachmentContenIds(after)));
+        // deleteAttachments(Collection<Attachment>) will delete an attachment that is present in `attachmentAfter`
+        Set<String> attachmentIdsAfter = getAttachmentContentIds(attachmentAfter);
+        Set<String> attachmentIdsToBeDeleted = new HashSet<>();
+
+        if(attachmentBefore != null) { // if the attachment prior to updating is null then it is either an invalid entry in the database or a test. In both case do nothing.
```

Comment:
That's a bunch of imperative code where it's not strictly necessary.
I think this code is neater and it passes the test.
```
        Set<Attachment> nonAcceptedAttachmentsBefore = nullToEmptySet(attachmentBefore).stream().filter(a -> a.getCheckStatus() != CheckStatus.ACCEPTED).collect(Collectors.toSet());
        deleteAttachmentsByIds(Sets.difference(getAttachmentContentIds(nonAcceptedAttachmentsBefore), getAttachmentContentIds(attachmentAfter)));

```